### PR TITLE
[Admin] [API] Improve product variant routing

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets |  |
| License         | MIT |

Without these fixes, we are accessing variants just based on their id.

### CODE REVIEW

File: src/Sylius/Bundle/AdminBundle/Resources/config/routing/product_variant.yml [hunk 1]
```diff
@@ -55,6 +55,11 @@ sylius_admin_product_variant_update:
                 route: sylius_admin_product_variant_index
                 parameters: { productId: $productId }
             permission: true
+            repository:
+                method: findOneByProductIdAndId
+                arguments:
+                    product: $productId
```
Comment (tuka217) [inline src/Sylius/Bundle/AdminBundle/Resources/config/routing/product_variant.yml]: productId

File: src/Sylius/Bundle/ApiBundle/Resources/config/routing/product_variant.yml [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-        _controller: sylius.controller.product_variant:indexAction
-        _sylius:
-            paginate: $limit
-            serialization_version: $version
-            grid: sylius_api_product_variant
-            section: api
-            serialization_groups: [Default]
-            criteria:
-                product: $productId
+sylius_api_product_variant:
```
Comment (tuka217) [inline src/Sylius/Bundle/ApiBundle/Resources/config/routing/product_variant.yml]: You should also remove prefix from main.yml file, and add this prefix to the rest of router paths.

File: src/Sylius/Bundle/ApiBundle/Resources/config/routing/product_variant.yml [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
+        serialization_version: $version
+        vars:
+            route:
+                parameters:
+                    productCode: $productCode
+    type: sylius.resource_api
 
 sylius_api_product_variant_create:
-    path: /
+    path: variants/
```
Comment (pamil) [inline src/Sylius/Bundle/ApiBundle/Resources/config/routing/product_variant.yml]: `variants/` -> `/variants`

File: src/Sylius/Bundle/ApiBundle/Resources/config/routing/product_variant.yml [hunk 4]
```diff
@@ -31,7 +31,7 @@ sylius_api_product_variant_create:
                   - expr:service('sylius.repository.product').findOneByCode($productCode)
 
 sylius_api_product_variant_update:
-    path: /{code}
+    path: variants/{code}
```
Comment (pamil) [inline src/Sylius/Bundle/ApiBundle/Resources/config/routing/product_variant.yml]: `variants/{code}` -> `/variants/{code}`

File: src/Sylius/Bundle/ApiBundle/Resources/config/routing/product_variant.yml [hunk 5]
```diff
@@ -40,26 +40,26 @@ sylius_api_product_variant_update:
...
             repository:
-                method: findOneByCode
-                arguments: [$code]
+                method: findOneByProductCodeAndCode
+                arguments: [$productCode, $code]
 
 
 sylius_api_product_variant_delete:
-    path: /{code}
+    path: variants/{code}
```
Comment (pamil) [inline src/Sylius/Bundle/ApiBundle/Resources/config/routing/product_variant.yml]: `variants/{code}` -> `/variants/{code}`

File: src/Sylius/Bundle/ApiBundle/Resources/config/routing/product_variant.yml [hunk 6]
```diff
@@ -40,26 +40,26 @@ sylius_api_product_variant_update:
...
-                method: findOneByCode
-                arguments: [$code]
+                method: findOneByProductCodeAndCode
+                arguments: [$productCode, $code]
             csrf_protection: false
 
 
 sylius_api_product_variant_show:
-    path: /{code}
+    path: variants/{code}
```
Comment (pamil) [inline src/Sylius/Bundle/ApiBundle/Resources/config/routing/product_variant.yml]: `variants/{code}` -> `/variants/{code}`

File: src/Sylius/Bundle/ProductBundle/Doctrine/ORM/ProductVariantRepository.php [hunk 7]
```diff
@@ -88,13 +88,31 @@ public function findByNameAndProduct($name, $locale, ProductInterface $product)
     /**
      * {@inheritdoc}
      */
-    public function findOneByCode($code)
```
Comment (pamil) [inline src/Sylius/Bundle/ProductBundle/Doctrine/ORM/ProductVariantRepository.php]: Needs an entry in `UPGRADE.md`.

File: UPGRADE.md [hunk 8]
```diff
@@ -268,6 +268,8 @@ After:
...
 * `ProductAssociationType::$name` property (and corresponding getter and setter) was removed to make it translatable. Therefore, `ProductAssociationTypeTranslation` was introduced with one `$name` property. All product association types names are migrated to new concept with migration `Version20161219160441`. Look at [this PR](https://github.com/Sylius/Sylius/pull/7134) if you have any problems with upgrade.
 * `Product::$availableOn` and `Product::$availableUntil` properties (and corresponding getters and setters) were removed. Look at [this PR](https://github.com/Sylius/Sylius/pull/7451) if you have any problems with upgrade.
+* `->findOneByCode()` has been replaced with `->findOneByCodeAndProductCode()` on `ProductVariantRepository`. The `$code` remains as a first argument, but the `$productCode` - new required argument has been added.
```
Comment (pamil) [inline UPGRADE.md]: ProductVariantRepository**Interface**

Comment (pamil) [inline UPGRADE.md]: Why don't we also specify the arguments of these methods?

**Comment by pjedrzejewski** [comment]
Thanks Łukasz!

**Pull request merged**